fastcrypto/src/encoding.rs

```diff
@@ -266,3 +267,22 @@ impl<'de, const N: usize> DeserializeAs<'de, [u8; N]> for Base58 {
         Ok(array)
     }
 }
+
+/// Bech32 encoding
+pub struct Bech32(String);
+
+impl Bech32 {
+    pub fn decode(s: &str, hrp: &str) -> Result<Vec<u8>, eyre::Report> {
```

Comment:
I'm guessing that hrp=Human Readable Part.
Let's add a comment about that, with examples.

Reply:
revised